src/rum/instrumentation/DdRumResourceTracking.tsx

```diff
@@ -5,14 +5,66 @@
  */
 
 import { DdRum } from '../../foundation'
-import type { DdRumXhr } from './DdRumXhr'
+import type { DdRumXhr, DdRumXhrContext } from './DdRumXhr'
 import { generateTraceId } from './TraceIdentifier';
 
 export const TRACE_ID_HEADER_KEY = "x-datadog-trace-id"
 export const PARENT_ID_HEADER_KEY = "x-datadog-parent-id"
 export const ORIGIN_HEADER_KEY = "x-datadog-origin"
 export const ORIGIN_RUM = "rum"
 
+const MISSING_TIME = -1
+
+type Duration = number & { d: 'Duration in ms' }
+type ServerDuration = number & { s: 'Duration in ns' }
+
+interface Timing {
+  startTime: ServerDuration
+  duration: ServerDuration
+}
+
+interface ResourceTimings {
+  // unlike in Performance API it is not the time until request
+  // starts (requestStart, before it can be connect, SSL, DNS),
+  // but the time until the response is first seen
+  firstByte: Timing,
+  download: Timing,
+}
+
+function createTimings(requestContext: DdRumXhrContext, responseEndTime: number): ResourceTimings | null {
```

Comment:
why not just passing the `startTime` and `loadStartTime` as arguments in this method signature. I think is much more cleaner this way.

Reply:
remainings of something bigger I wanted to implement, but failed because of missing `Resource Timing API`. Indeed with only `firstByte` and `download` timings we may simplify that.